packages/wrangler/src/pages.tsx

```diff
@@ -830,6 +830,8 @@ export const pages: BuilderCallback<unknown, unknown> = (yargs) => {
             scriptPath,
           };
         } else {
+          // TODO: this might throw because
+          // it's used before being defined
           scriptReadyResolve();
```

Comment:
This is a weird one. I dunno how (or if) it even works.

Reply:
The problem here is that TypeScript has no way of knowing when the callback passed to the `Promise` constructor gets called.

```
const scriptReadyPromise = new Promise(
  (resolve) => (scriptReadyResolve = resolve)
);
```

We happen to know that the semantic of the Promise constructor, is that it will call the callback synchronously (see https://tc39.es/ecma262/multipage/control-abstraction-objects.html#sec-promise-executor). So we can be sure that `scriptReadyResolve` is well defined.

So the solution here is to use the `!` not null assertion operator, and a suitable eslint pragma.

Reply:
Thanks!